pkg/registry/etcd/ns_server.go

```diff
@@ -59,10 +59,10 @@ func (n *etcdNSRegistryServer) Register(ctx context.Context, request *registry.N
 	)
 	if apierrors.IsAlreadyExists(err) {
 		var exist *v1.NetworkService
-		exist, err = n.client.NetworkservicemeshV1().NetworkServices(n.ns).Get(ctx, request.Name, metav1.GetOptions{})
+		exist, err = n.client.NetworkservicemeshV1().NetworkServices("").Get(ctx, request.Name, metav1.GetOptions{})
 		if err == nil {
 			exist.Spec = *(*v1.NetworkServiceSpec)(request)
-			apiResp, err = n.client.NetworkservicemeshV1().NetworkServices(n.ns).Update(ctx, exist, metav1.UpdateOptions{})
```

Comment:
Finally I think we should keep `n.ns` for Create and Update.

Reply:
ok, done